pkg/covermerger/repos.go

```diff
@@ -40,10 +44,25 @@ func getFileVersions(c *Config, targetFilePath string, rbcs []RepoBranchCommit,
 
 type repoCache struct {
 	cache map[RepoBranchCommit]vcs.Repo
+	mu    sync.RWMutex
+}
```

Comment:
Overall, the implementation `repoCache` type looks somewhat strange -- why keep multiple `vcs.Repo` instances that all point to the same folder? They are all effectively the same, even if they are obtained via separate calls to `vcs.NewRepo`.

Reply:
What is your proposal?
Logical representation and physical may be different.

Reply:
You just need to keep track of what RepoBranchCommt have already been polled (and serialize the polls which have to be done), you don't need many separate `vcs.Repo` instances.

Reply:
Do we even need to keep track of what RepoBranchCommt have already been polled?
If we are aggregating 10 repos, I would assume we can fetch all of them into a single repo first and then just do "git show" in parallel.

Reply:
The code keep track of what RepoBranch was polled. It means I have 1 vcsRepo for every RepoBranch.
I can pre-poll all the repositories but it will require the list or repos.
The list can be received from bigquery or from the stream.
Getting it from the stream means 1 more stream processing pass and more parameters for the tool(repo list).
Getting it from the bigquery means 1 more bigquery request and more parameters for the tool(repo list).

I don't think repo prefetching will simplify code.
Current implementation is a stream processing. And the stream defines what repos are needed.

Reply:
I can store 1 vcs.Repo pointer instead of 10-20-30 but I'm not sure the code will be simpler.

Reply:
Let me try.

Reply:
Does it look better to you?

Reply:
To close this thread:
1. repos.go rewritten. It looks better to me now.
2. investing more time I can rewrite syz-bq part in golang and add prefetching. Adding it to the current code doesn't look good to me (from the RoI perspective).